tensorflow_datasets/core/features/video_feature.py

```diff
@@ -185,27 +177,53 @@ def to_json_content(self) -> Json:
         'ffmpeg_extra_args': self._extra_ffmpeg_args
     }
 
-  def _generate_video_html(self, images, framerate=24, encoding_format='mp4') -> str:
+  def _generate_video_html(self, images, framerate=24) -> str:
     """Converts sequence of images into video string."""
-    imgs = len(str(len(images)))+1  # Find number of digits in len to give names.
+    num_digits = len(str(len(images)))+1  # Find number of digits in len to give names.
 
-    with tempfile.TemporaryDirectory() as video_dir:
-      for i, img in enumerate(images):
-        f = os.path.join(video_dir, f'img{i:0{imgs}d}.png')
-        img.save(f, format='png')
+    try:
+      dummy_ffmpeg_args = [self._ffmpeg_path, '-version']
+      _ffmpeg_run(dummy_ffmpeg_args)
+
+    except IOError:
+      # if ffmpeg is not installed, generate GIF using pngs
+      def write_buff(buff):
+        images[0].save(
+            buff,
+            format='GIF',
+            save_all=True,
+            append_images=images[1:],
+            # Could add a frame_rate kwargs in __init__ to customize this.
+            duration=1000/framerate,  # 41ms / img ~= 24 img / sec
+            loop=0,
+        )
+
+      # Convert to base64
+      gif_str = utils.get_base64(write_buff)
+      return f'<img src="data:image/png;base64,{gif_str}" alt="Gif" />'
 
-      ffmpeg_args = [self._ffmpeg_path, '-framerate', str(framerate), '-i',
-                    os.fspath(os.path.join(video_dir, f'img%0{imgs}d.png'))]
+    else:
+      with tempfile.TemporaryDirectory() as video_dir:
+        for i, img in enumerate(images):
+          f = os.path.join(video_dir, f'img{i:0{num_digits}d}.png')
+          img.save(f, format='png')
 
-      extra_ffmpeg_args = ['-vf', 'scale=128:128', '-vcodec', 'h264',
-                           '-pix_fmt', 'yuv420p', '-allow_sw', '1']
-      ffmpeg_stdin = None
+        ffmpeg_args = [self._ffmpeg_path, '-framerate', str(framerate), '-i',
+                      os.fspath(os.path.join(video_dir, f'img%0{num_digits}d.png'))]
 
-      output_pattern = os.path.join(video_dir, 'output.' + encoding_format)
-      ffmpeg_args += extra_ffmpeg_args
-      ffmpeg_args.append(output_pattern)
-      try:
-        _ffmpeg_run(ffmpeg_args, ffmpeg_stdin)
+        extra_ffmpeg_args = [
+          '-vf', 'scale=128:128', # scale output videos to 128x128
+          # using native h264 encoder
+          '-vcodec', 'h264',      # output video stream codec - H.264
+          '-pix_fmt', 'yuv420p',  # use yuv420v pixel format (enhance compatibility)
```

Comment:
enhance compatibility with what ?

Reply:
When using H.264 coded video, it ensures that not so good players can decode the video. Please see:
https://trac.ffmpeg.org/wiki/Slideshow

Reply:
Thank you. Not everyone is familiar with FFMPEG, so I think we should make sure comments are explainatory. For example. we should add https://trac.ffmpeg.org/wiki/Slideshow to the comment